### PR TITLE
Adjust Av and Age priors, allow verify_params failures

### PR DESCRIPTION
**Pull request by lcjohnso**
Here I enable three optional features:
1) Add an exponential Av prior model.
2) Allow the removal of the "constant in linear age" prior to prefer the grid's intrinsic logarithmic spacing.
3) Allow verify_params failures to print to screen, but continue without exiting.  This is helpful when using non-fA dust curves and fixed R_V values.

While not included in this PR, I would be happy to add info about what needs to be changed in run_beast.py and datamodel.py to use these updates, but I'll wait until the datamodel.py documentation page is drafted (Issue https://github.com/BEAST-Fitting/beast/issues/71).

### CODE REVIEW

**Comment by karllark** [comment]
In the verify_params changes, I'm not sure having the exit being overridden for all possible exits is a good idea.  In fact, ideally the override would not be done at all, but the verify_params logic changed to handle the two cases mentioned - no fA and/or fixed Rv extinction curves.  @lcjohnso: Is such a change in the logic not possible/straightforward?

**Comment by lcjohnso** [comment]
Re: @karllark's suggestion, I added logic to verify_params that allows for non-`exit()` warnings in the case that:

1) `fAs = None`, as needed for case where non-Gordon16 extinction curve is used.

2) Parameter grids are defined as single values (e.g., defining sequence is [0.0, 0.0, 0.1] leading to single grid point at 0.0).

I'm leaving the `noexit` option implemented (could still have use in future), but its use is discouraged as it might allow non-physical grid/parameter values.

**Comment by lcjohnso** [comment]
Also: I added initial documentation of datamodel.py/run_beast.py, including HowTo instructions for implementing new optional features.

This addition is a good step towards closing Issue https://github.com/BEAST-Fitting/beast/issues/71 -- please comment there on additions that might still be needed before closing issue.

File: beast/physicsmodel/prior_weights_dust.py [hunk 1]
```diff
@@ -85,6 +85,23 @@ def _two_lognorm(xs,
...
 
+def _exponential(x, a=2.0, N=1.):
+    """
+    Exponential distribution
+    Parameters
+    ----------
+    x: vector
+       x values
+    a: float
+       Single parameter in exp: a*e^-ax
```
Comment (karllark) [inline beast/physicsmodel/prior_weights_dust.py]: This should be N*e^-ax.

File: beast/physicsmodel/prior_weights_dust.py [hunk 2]
```diff
@@ -85,6 +85,23 @@ def _two_lognorm(xs,
...
+    N: floats
```
Comment (karllark) [inline beast/physicsmodel/prior_weights_dust.py]: Is this parameter needed?  Shouldn't this be set by the desire to have this prior integrate to 1 over the min/max age range?  If so, then it is not needed and should be set inside the function.

File: beast/tools/verify_params.py [hunk 3]
```diff
@@ -13,39 +13,43 @@
...
-        exit()
+        if not noexit: exit()
 
     if (param_max-param_min) < param_step:
-        print((param_name + ' step value greater than (max-min)'))
-        exit()
-
-def verify_one_input_format(param, param_name, param_format, param_lim):
+        if param_max-param_min == 0.:
+            print('Warning: '+param_name+' grid is single-valued.')
```
Comment (karllark) [inline beast/tools/verify_params.py]: Nit picky point.  I don't think this is a warning.  There is no problem with single value parameters in a grid.  I worry that having it issue a warning makes it sound like this is bad.  Maybe something less "scary" like "Info"?

File: docs/beast_setup.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
+Running the BEAST
```
Comment (karllark) [inline docs/beast_setup.rst]: Great to have this doc file!  Many thanks for starting it.  I know there is material I will add once I think about it more.

File: beast/tools/verify_params.py (hunk 3 above)
Comment (lcjohnso) [reply beast/tools/verify_params.py]: Changed to "Note:"

File: beast/physicsmodel/prior_weights_dust.py (hunk 2 above)
Comment (lcjohnso) [reply beast/physicsmodel/prior_weights_dust.py]: I was following the format of the other prior choices.  Will leave as is.

File: beast/physicsmodel/prior_weights_dust.py (hunk 1 above)
Comment (lcjohnso) [reply beast/physicsmodel/prior_weights_dust.py]: I corrected the notation, but am leaving the param as "a" -- could use "decay_rate" or "decay_param" but that seems overkill.  Instead, I added additional text in docstring.

**Pull request merged**